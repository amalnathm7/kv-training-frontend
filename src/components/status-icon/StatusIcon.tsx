import { StatusType } from "../../types/StatusType";
import React from "react";

export type StatusIconPropsType = {
    status: StatusType
};

const StatusIcon: React.FC<StatusIconPropsType> = (props) => {
    const style = {
        backgroundColor: props.status.color,
        paddingTop: "5px",
        paddingBottom: "5px",
        paddingLeft: "20px",
        paddingRight: "20px",
        borderRadius: "20px",
        width: "100px",
        display: "flex",
        justifyContent: "center",
        fontSize: "12px"
    };

    return <div style={style} className="status-container">
        <label>{props.status.label}</label>
    </div>;
};

export default StatusIcon;
