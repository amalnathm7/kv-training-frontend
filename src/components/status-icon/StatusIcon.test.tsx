import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import StatusIcon, { StatusIconPropsType } from './StatusIcon';

describe('StatusIcon Props Test', () => {
  test('If StatusIcon rendered correctly', () => {
    const props: StatusIconPropsType = {
      status: {
        label: 'Status-label',
        color: '#000000'
      }
    };

    const element = render(<StatusIcon {...props} />);

    expect(element).toMatchSnapshot();
  });
});
