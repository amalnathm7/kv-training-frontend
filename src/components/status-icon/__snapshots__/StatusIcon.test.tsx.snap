// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusIcon Props Test If StatusIcon rendered correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="status-container"
        style="background-color: rgb(0, 0, 0); padding: 5px 20px 5px 20px; border-radius: 20px; width: 100px; display: flex; justify-content: center; font-size: 12px;"
      >
        <label>
          Status-label
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="status-container"
      style="background-color: rgb(0, 0, 0); padding: 5px 20px 5px 20px; border-radius: 20px; width: 100px; display: flex; justify-content: center; font-size: 12px;"
    >
      <label>
        Status-label
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
